library_generation/generate_composed_library.py

```diff
@@ -0,0 +1,141 @@
+"""
+This script allows generation of libraries that are composed of more than one
+service version. It is achieved by calling `generate_library.sh` without
+postprocessing for all service versions and then calling
+postprocess_library.sh at the end, once all libraries are ready.
+
+Prerequisites
+- Needs a folder named `output` in current working directory. This folder
+is automatically detected by `generate_library.sh` and this script ensures it
+contains the necessary folders and files, specifically:
+  - A "google" folder found in the googleapis/googleapis repository
+  - A "grafeas" folder found in the googleapis/googleapis repository
+Note: googleapis repo is found in https://github.com/googleapis/googleapis.
+"""
+
+import click
+import utilities as util
+import os
+import sys
+import subprocess
+import json
+from model.GenerationConfig import GenerationConfig
+from model.Library import Library
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+
+"""
+Main function in charge of generating libraries composed of more than one
+service or service version.
+Arguments
+ - config: a GenerationConfig object representing a parsed configuration
+ yaml
+ - library: a Library object contained inside config, passed here for
+   convenience and to prevent all libraries to be processed
+ - enable_postprocessing: true if postprocessing should be done on the generated
+   libraries
+ - repository_path: path to the repository where the generated files will be
+   sent. If not specified, it will be downloaded. The versions file will be inferred from this folder
+"""
+def generate_composed_library(
+    config: GenerationConfig,
+    library: Library,
+    enable_postprocessing: bool,
+    repository_path: str
+) -> None:
+  output_folder = util.sh_util('get_output_folder')
+
+  print(f'output_folder: {output_folder}')
+  print('library: ', library)
+  os.makedirs(output_folder, exist_ok=True)
+
+  googleapis_commitish = config.googleapis_commitish
+  if library.googleapis_commitish is not None:
+    googleapis_commitish = library.googleapis_commitish
+    print('using library-specific googleapis commitish: ' + googleapis_commitish)
+  else:
+    print('using common googleapis_commitish')
+
+  print('removing old googleapis folders and files')
+  util.delete_if_exists(f'{output_folder}/google')
+  util.delete_if_exists(f'{output_folder}/grafeas')
+
+  print('downloading googleapis')
+  util.sh_util(f'download_googleapis_files_and_folders "{output_folder}" "{googleapis_commitish}"')
+
+
+  base_arguments = []
+  base_arguments += util.create_argument('gapic_generator_version', config)
+  base_arguments += util.create_argument('grpc_version', config)
+  base_arguments += util.create_argument('protobuf_version', config)
+
+  destination_path = f'java-{library.api_shortname}'
+  if config.destination_path == 'google-cloud-java':
```

Comment:
How about change the str constant to a variable?

Reply:
Changed to a constant at the beginning of the file. I also simplified a coouple of `if`s to use an `is_monorepo` variable